tests/perception/high_level_semantics_situation_to_developmental_primitive_perception_test.py

```diff
@@ -40,3 +40,29 @@ def test_person_and_ball():
     assert person_and_ball_perception.frames[0].relations
     print(person_and_ball_perception.frames[0].property_assertions)
     assert len(person_and_ball_perception.frames[0].property_assertions) == 2
+
+
+def test_person_and_ball_color():
+    person = SituationObject(PERSON)
+    ball = SituationObject(BALL, properties=[RED])
+
+    person_and_ball_situation = HighLevelSemanticsSituation(
+        ontology=GAILA_PHASE_1_ONTOLOGY, objects=[person, ball]
+    )
+
+    person_and_ball_perception = _PERCEPTION_GENERATOR.generate_perception(
+        person_and_ball_situation, chooser=RandomChooser.for_seed(0)
+    )
+
+    assert len(person_and_ball_perception.frames) == 1
+    assert len(person_and_ball_perception.frames[0].property_assertions) == 3
+
+    property_handles = set(
+        str(assertion)
```

Comment:
To clarify my earlier comment a bit: you don't want this test to start crashing because someone changes the `str` method of a class.

So in this case, you might want something like (off the top of my head without checking the field names, etc.):

```
property_assertions = person_and_ball_perception.frames[0].property_assertions
assert (prop_assertion is PropertyAssertion for prop_assertion in property_assertions)
property_pairs = {(prop_assertion.arg.handle,prop_assertion.property.handle
                                              for prop_assertion in property_assertions})
assert arg_and_prop_strings == {("person_0", "animate"), ("ball_0", "inanimate")}
...
```

Reply:
I think the problem with this is that the objects are not the same type - HasColor has the color field, while HasBinaryProperty has binary_property field. So I will have to write a test that checks for the type first.

Reply:
@denizbeser : You might find this useful for that purpose:
https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.partition

Reply:
Thanks! I already fixed it with isinstance(). Pushing now!